src/vmm/src/linux/vstate.rs

```diff
@@ -624,13 +636,15 @@ impl Vm {
                     .map_err(Error::SetUserMemoryRegion)?;
             };
 
+            #[cfg(not(target_arch = "riscv64"))]
```

Comment:
aarch64 busy-loops with guest memfd currently, could that be what you've been seeing on riscv64 as well?